msal/src/main/java/com/microsoft/identity/client/MultipleAccountPublicClientApplication.java

```diff
@@ -289,52 +332,116 @@ public void onError(final Exception exception) {
+        params.setAccount(requestAccountRecord);
+
         try {
-            if (null == multiTenantAccount) {
-                com.microsoft.identity.common.internal.logging.Logger.warn(
-                        TAG,
-                        "Requisite IAccount or IAccount fields were null. Insufficient criteria to remove IAccount."
-                );
-
-                callback.onTaskCompleted(false);
-            } else {
-                final OperationParameters params = OperationParametersAdapter.createOperationParameters(mPublicClientConfiguration);
-
-                // TODO Clean this up, only the cache should make these records...
-                // The broker strips these properties out of this object to hit the cache
-                // Refactor this out...
-                final AccountRecord requestAccountRecord = new AccountRecord();
-                requestAccountRecord.setEnvironment(multiTenantAccount.getEnvironment());
-                requestAccountRecord.setHomeAccountId(multiTenantAccount.getHomeAccountId());
-                params.setAccount(requestAccountRecord);
-
-                final RemoveAccountCommand command = new RemoveAccountCommand(
-                        params,
-                        MSALControllerFactory.getAcquireTokenSilentControllers(
-                                mPublicClientConfiguration.getAppContext(),
-                                params.getAuthority(),
-                                mPublicClientConfiguration
-                        ),
-                        callback
-                );
-
-                ApiDispatcher.removeAccount(command);
-            }
+            final RemoveAccountCommand command = new RemoveAccountCommand(
+                    params,
+                    MSALControllerFactory.getAcquireTokenSilentControllers(
+                            mPublicClientConfiguration.getAppContext(),
+                            params.getAuthority(),
+                            mPublicClientConfiguration
+                    ),
+                    new TaskCompletedCallbackWithError<Boolean, BaseException>() {
+                        @Override
+                        public void onError(BaseException error) {
+                            callback.onError(MsalExceptionAdapter.msalExceptionFromBaseException(error));
+                        }
+
+                        @Override
+                        public void onTaskCompleted(Boolean success) {
+                            callback.onRemoved();
+                        }
+                    }
+            );
+
+            ApiDispatcher.removeAccount(command);
+
         } catch (final MsalClientException e) {
             callback.onError(e);
         }
     }
 
+    @Override
+    public boolean removeAccount(@Nullable IAccount account) throws MsalException, InterruptedException {
+
+        final ResultFuture<RemoveAccountResult> future = new ResultFuture();
```

Comment:
AsyncResult?

Reply:
I didn't in this case since there is not result object.... I suppose I can use Boolean.  I'll update.

Reply:
Fixed.